poetry/core/masonry/builders/builder.py

```diff
@@ -364,17 +363,12 @@ def convert_author(cls, author: str) -> Dict[str, str]:
     @classmethod
     @contextmanager
     def temporary_directory(cls, *args: Any, **kwargs: Any) -> ContextManager[str]:
-        try:
-            from tempfile import TemporaryDirectory
-
-            with TemporaryDirectory(*args, **kwargs) as name:
-                yield name
-        except ImportError:
-            name = tempfile.mkdtemp(*args, **kwargs)
+        from tempfile import TemporaryDirectory
 
+        with TemporaryDirectory(*args, **kwargs) as name:
             yield name
 
-            shutil.rmtree(name)
+        shutil.rmtree(name)
```

Comment:
This change is strictly speaking incorrect -- the TemporaryDirectory contextmanager will clean up on its own. Removal of the entire wrapper is better than removing this line however.

Reply:
It looks like `rmtree` line is still around.